README.md
# map_provider

A plugin for showing static map and calling Google Map application when clicked

## Getting Started

Add into your pubspec.yml
``` dart
dependencies:
  map_provider: ^1.0.0
```
Then install
``` dart
$ flutter packages get
```
Add to your project
``` dart
import 'package:map_provider/map_provider.dart';
```

### Simple HOW-TO
``` dart
(...)
import 'package:map_provider/map_provider.dart;

class MyMapView extends StatefulWidget{
  (...)
}

class _MyMapViewState extends State<MyMapView>{
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Plugin example app'),
        ),
        body: Center(
          child: Container(
              height: 200,
              width: 200,
              child: MapProvider(
                google_api_key: '' , ///your_api_key_here, otherwise this won\'t work
                width: 200,
                height: 200,
                latitude: '15.79286',
                longitude: '27.94013',
                zoom: 1,
              )
          ),
        ),
      ),
    );
  }
}
```


CHANGELOG.md
## 1.0.1

* Change Readme and deleting unused package

## 1.0.0

* Initial Release


test/map_provider_test.dart
//import 'package:flutter/services.dart';
//import 'package:flutter_test/flutter_test.dart';
//import 'package:map_provider/map_provider.dart';
//
//void main() {
////  const MethodChannel channel = MethodChannel('map_provider');
////
////  setUp(() {
////    channel.setMockMethodCallHandler((MethodCall methodCall) async {
////      return '42';
////    });
////  });
////
////  tearDown(() {
////    channel.setMockMethodCallHandler(null);
////  });
////
////  test('getPlatformVersion', () async {
////    expect(await MapProvider.platformVersion, '42');
////  });
//}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:map_provider_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that platform version is retrieved.
    expect(
      find.byWidgetPredicate(
        (Widget widget) => widget is Text &&
                           widget.data.startsWith('Running on:'),
      ),
      findsOneWidget,
    );
  });
}


lib/map_provider.dart
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';

class MapProvider extends StatefulWidget{

  final String google_api_key;
  final String latitude;
  final String longitude;
  final int width;
  final int height;
  final int zoom;

  const MapProvider({Key key, @required this.google_api_key, this.latitude, this.longitude, this.width, this.height, this.zoom}) : super(key: key);

  @override
  State<StatefulWidget> createState() {
    return _MapProviderState();
  }
}

class _MapProviderState extends State<MapProvider>{
  @override
  Widget build(BuildContext context) {
    //First thing first, we need to initialize the URL For the static map and the GoogleMap
    String mapURI = 'https://maps.googleapis.com/maps/api/staticmap?center=${widget.latitude}%2C${widget.longitude}&zoom=${widget.zoom}&size=${widget.width}x${widget.height}&maptype=roadmap&markers=color:green%7C${widget.latitude},${widget.longitude}&key=${widget.google_api_key}';
    String googleMap = 'https://www.google.com/maps/search/?api=1&query=${widget.latitude},${widget.longitude}';

    return GestureDetector(
      onTap: (){
        openMap(googleMap, context); ///here we call the method for navigating to Google Map application
      }, /// We need to use GestureDetector, so we can click on the map image later
      child: Image.network(mapURI, fit: BoxFit.fill,), ///Show the image from the staticMap url.
    );
  }

  ///create method
  static openMap(String googleMapUrl, BuildContext context) async {
    String url = googleMapUrl; //initialize url
    if(await canLaunch(url)){ ///check if we can launch the url
      await launch(url); ///then launch the url
    }
    else{
      ///otherwise show snackbar error message for 3 seconds
      Scaffold.of(context).showSnackBar(SnackBar(
        backgroundColor: Colors.red,
        content: Text('Couldn\'t open map'),
        duration: Duration(seconds: 3),
      ));
    }
  }
}


android/src/main/java/org/digidaw/map_provider/MapProviderPlugin.java
package org.digidaw.map_provider;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** MapProviderPlugin */
public class MapProviderPlugin implements MethodCallHandler {
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "map_provider");
    channel.setMethodCallHandler(new MapProviderPlugin());
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else {
      result.notImplemented();
    }
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:map_provider/map_provider.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Plugin example app'),
        ),
        body: Center(
          child: Container(
              height: 200,
              width: 200,
              child: MapProvider(
                google_api_key: '' , ///your_api_key_here, otherwise this won\'t work
                width: 200,
                height: 200,
                latitude: '15.79286',
                longitude: '27.94013',
                zoom: 1,
              )
          ),
        ),
      ),
    );
  }
}
